contracts/ERC20Safe.sol

```diff
@@ -291,4 +316,24 @@ contract ERC20Safe is BridgeRole, Pausable {
         Batch memory batch = batches[batchesCount - 1];
         return _isBatchProgessOver(batch.depositsCount, batch.blockNumber) || batch.depositsCount >= batchSize;
     }
+
+    function _internalMint(address token, address to, uint256 amount) internal returns (bool) {
+        IMintableBurnableERC20 mintableToken = IMintableBurnableERC20(token);
+
+        if (!mintableToken.hasRole(mintableToken.MINTER_ROLE(), address(this))) {
+            return false;
+        }
+
+        try mintableToken.mint(to, amount) {
+            return true;
+        } catch {
+            return false;
+        }
+    }
+
+    function _internalBurn(address token, address from, uint256 amount) internal {
```

Comment:
Like I said above, we'd better use existing audited contracts here. Better for ecosystem fit also.
As a note, if projects would deploy the exact version of ERC20 contracts we have here, burning would underflow. We should update the library version and use the versions there, where we have everything we need, also underflow checks: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L188